src/pybamm/config.py

```diff
@@ -110,7 +116,7 @@ def get_input():  # pragma: no cover
 
 
 def generate():
-    if is_running_tests():
+    if is_running_tests() or check_opt_out():
```

Comment:
Since `is_running_tests()` does not work in all CI cases, I put the opt-out check here to provide another escape